tictactoe.rb

```diff
@@ -0,0 +1,128 @@
+class Game
+    attr_accessor :board, :board_o, :board_x, :b
+
+    def initialize
+        #to intialize -start method and winning positions
+        start
+       
+        @win_positions =
+      [[0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7], [2,5,8], [0,4,8], [2,4,6]]
+    end
+
+    def start
+        #@turns_played = 0
+         @player = "x"
+        @board = ["-", "-", "-", "-", "-", "-", "-", "-", "-"]
+        @oard_o=0
+        @board_x=0
+        @b=0
+    end
+    def show_board(board = @board)
+        puts board[6..8].join(" ")
+        puts board[3..5].join(" ")        
+        puts board[0..2].join(" ")
+    end
+
+    def play
+        @play_game = true
+        while @play_game
+            puts
+             puts "Tic Tac Toe"          
+            show_board
+            check_win
+            check_draw
+            turn
+            change_player
+        end
+      
+    end 
+    
+    def end_game        
+        puts "The game has finished!!"
+        exit
+    end
+    
+    def check_win 
+        @board_x=0 
+        @board_o=0 
+        #checking each position in the array x inside winnig positions
+        @win_positions.each do |x|
```

Comment:
What is `@win_positions`? An array? This would read better if you replaced `x` for `row`, so it will quickly be interpreted as **for row in winning_positions**.